common/src/main/java/com/wynntils/models/emeralds/EmeraldModel.java

```diff
@@ -153,7 +156,13 @@ public void onSetSlot(SetSlotEvent.Pre event) {
         if (pouchContainerId != -1 && !isInventory) return;
 
         // Subtract the outgoing object from our balance
-        adjustBalance(event.getContainer().getItem(event.getSlot()), -1, isInventory);
+        ItemStack outgoingItemStack = event.getContainer().getItem(event.getSlot());
+        if (handledEmeraldStacks.containsKey(event.getSlot())) {
+            // We've already handled this stack, replace the outgoingItemStack with the one we've already seen
+            // to force the balance to subtract correctly
+            outgoingItemStack = handledEmeraldStacks.get(event.getSlot());
+        }
+        adjustBalance(outgoingItemStack, -1, isInventory);
```

Comment:
what? We remove it then add it the next line? Shouldn't this be in the if block?

Is `event.getItemStack()` the same as `event.getContainer().getItem(event.getSlot())`? If so, please use the first one.

Reply:
`event.getContainer().getItem(event.getSlot())` is the OLD item.. Okay. Can you rename `outgoingItemStack` because I could not figure this out on reading

Reply:
Removing and adding it in the next line was the original behaviour. If you just put a print statement in the adjustBalance method at the end of the file, there is actually a lot of adding, removing, setting to zero, adding again, etc. If we want to fix that, I think that would be better in a separate PR

And no, the two event methods are not the same. .getItemStack() returns the new item being inputted into that slot, where the getContainer() way returns the old item being replaced.

Reply:
> `event.getContainer().getItem(event.getSlot())` is the OLD item.. Okay. Can you rename `outgoingItemStack` because I could not figure this out on reading

Okay, I see we did not change the original code in this way.. But still, please rename this,  then we are good to go.